Sources/StreamChatUI/ChatMessageList/ChatTypingIndicatorView.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+// TODO: Decide which IndicatorView to Use:
+// A) AChatTypingIndicatorView
+// B) ChatTypingIndicatorView
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+public typealias AChatTypingIndicatorView = _ChatTypingIndicatorView<NoExtraData>
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+public typealias ChatTypingIndicatorView = _ChatTypingIndicatorView<NoExtraData>
```

Comment:
IMHO the `ChatTypingIndicatorView` is much better looking + memory friendly. It doesn't use any complicated API and the layer is easily replicated. Also it animates properly, the `AChatTypingIndicatorView` sometimes doesn't finish the animationGroup and behaves a lot strange..

I tried to implement this via `UIViewPropertyAnimator` but there is no simple way of repeating the animations, hence read this: https://developer.apple.com/forums/thread/122209

Reply:
Please let me know which of the 2 indicators you prefer ^^